### ISSUE TITLE
How to disable keystrokes when focus on input, textarea, select?

### ISSUE THREAD

**Issue by maximelebreton** [RobertWHurst/Keystrokes#29]
Hello, and thanks for your work!

I would like to know how disable keystrokes when the focus is on an input, a textarea or a select, like the filter function of hotkeysjs: https://github.com/jaywcjlove/hotkeys-js#filter

Thanks

**Comment by RobertWHurst** [comment]
Ah yes. You've stumbled into a bit of an issue with the test stub I've created. It doesn't implement the browser bindings keystrokes uses by default. I'll come up with a solution for this soon.

**Comment by RobertWHurst** [comment]
My suggestion would be to create a utility function to filter these events.

```ts
import { KeyComboEvent, bindKeyCombo } from "@rwh/keystrokes";
import { BrowserKeyComboEventProps } from "@rwh/keystrokes/dist/browser-bindings";

// NOTE: This will be added to Keystrokes in the next release
type BrowserKeyComboEvent = KeyComboEvent<KeyboardEvent, BrowserKeyComboEventProps, {}>

const shouldFilter = (e: BrowserKeyComboEvent) => {
  const browserEvent = e.finalKeyEvent.originalEvent
  const target = (browserEvent.target ?? browserEvent.srcElement) as HTMLElement;
  const tagName = target.tagName;

  return (
    target.isContentEditable ||
    tagName == 'INPUT' ||
    tagName == 'SELECT' ||
    tagName == 'TEXTAREA'
  )
}

bindKeyCombo('a + b', (e) => {
  if (shouldFilter(e)) return

  console.log('Do something cool, but only if not in an input')
});
```

The main annoyance here is the fact that it's a bit cumbersome to type out the event type. I plan on fixing these for TypeScript users in the next release by including convenance types for browser binding associated key event and key combo events.

**Comment by RobertWHurst** [comment]
Ok I've just published 1.4.0, and it contains BrowserKeyComboEvent one does not need to define it themselves now.

**Comment by RobertWHurst** [comment]
Going to close this issue for now, but let me know if you have thoughts on this.
